### PR TITLE
peviitor_scraper refactoring

### PR DESCRIPTION
**Pull request by VACostin**
The main goal of this refactoring process is to streamline the future work of .js scrapers.

### CODE REVIEW

**Comment by VACostin** [comment]
Is there an environment where I can test these changes?

File: peviitor_scraper.js [hunk 1]
```diff
@@ -2,33 +2,33 @@
...
+        scraper.headers.headers["Content-Type"] = "application/json";
 
-          scraper.post(JSON.stringify(data));
-        });
+        scraper.post(JSON.stringify(data));
+      });
     })
     .catch((error) => {
-      console.log("Error sending trigger for " + file);
+      console.log(`Error sending trigger for ${file}`);
```
Comment (VACostin) [inline peviitor_scraper.js:31]: Todo: Find out where is file initialized

File: lib/scraper.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        "User-Agent":
+          "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.132 Safari/537.36",
+      },
+    };
+    this.soup = this.get_soup();
+  }
+
+  async get_soup() {
+    const response = await axios.get(this.url, this.headers);
+    return new Jssoup(response.data);
```
Comment (VACostin) [inline lib/scraper.js:20]: is this used anywhere?

File: lib/api_scraper.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+class ApiScraper {
+  constructor(url) {
+    this.url = url;
+    this.user_agent =
+      "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.132 Safari/537.36";
+    this.headers = {
+      headers: {
+        "User-Agent": this.user_agent,
+      },
+    };
```
Comment (VACostin) [inline lib/api_scraper.js:14]: How about using
```javascript
this.config = {
    headers: {
       "etc"
    }
}
```
this instead, it's less confusing

Comment (VACostin) [reply lib/api_scraper.js:14]: I'm aware this change breaks things (spoiler, it's not going to be the last one), which is good, it forces us to do 2 things:

1. We force a consistent coding style
2. More unit tests!

**Comment by VACostin** [comment]
Todos for tomorrow:

My biggest pet peeve for which I'm gonna use my first scraper example:
https://github.com/peviitor-ro/based_scraper_js/blob/8839365dab6f9f46309100af0c1770b7153ee16a/sites/syncrosoft.js#L43

After calling something like publish(bigObject, smallerObject) our job should be done. The user shouldn't have to rewrite the publish function each time.

Also, we should generate the id ourselves and let the user handle just the scraping part.
https://github.com/peviitor-ro/based_scraper_js/blob/8839365dab6f9f46309100af0c1770b7153ee16a/sites/syncrosoft.js#L36

Extra brownie points: https://github.com/peviitor-ro/based_scraper_js/blob/8839365dab6f9f46309100af0c1770b7153ee16a/sites/syncrosoft.js#L4
Let's trim the whitespaces and let the user put as many whitespaces as he can

+some other minor things.

**Review by lalalaurentiu — DISMISSED** [review]

**Pull request closed without merging**